### PR TITLE
Feature/4996 tsk data donation

### PR DESCRIPTION
**Pull request by SwissMountainFrog**
This PR aggregates the Delta Onboarding for data donation & onboarding for data donation.

[JIRA Ticket 4788](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-4788)

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/Datadonation/DataDonationViewController.swift [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+	
+	func wasAttemptedToBeDismissed() {
+		
+	}
+
+	// MARK: - Public
+
+	// MARK: - Internal
+	
+	var finished: (() -> Void)?
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Scenes/Datadonation/DataDonationViewController.swift:88]: Can you add some documentation please?

File: src/xcode/ENA/ENA/Source/Scenes/Datadonation/DataDonationViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+//		return dataDonationModel.federalStateName ?? AppStrings.DataDonation.Info.noSelectionState
+//	}
+//
+//	private var friendlyRegionName: String {
+//		return dataDonationModel.region ?? AppStrings.DataDonation.Info.noSelectionRegion
+//	}
+//
+//	private var friendlyAgeName: String {
+//		return dataDonationModel.age ?? AppStrings.DataDonation.Info.noSelectionAgeGroup
+//	}
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Scenes/Datadonation/DataDonationViewModel.swift:156]: Still needed?

**Review by pascal-brause — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/Datadonation/District.swift [hunk 3]
```diff
@@ -1,25 +1,28 @@
-//
+////
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Scenes/Datadonation/District.swift:1]:
```suggestion
//
```

File: src/xcode/ENA/ENA/Source/Scenes/Datadonation/__tests__/DataDonationModelTests.swift [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+import XCTest
+@testable import ENA
+
+class DataDonationModelTests: XCTestCase {
+
+	func testGIVEN_ModelWithJson_WHEN_GetValues_THEN_ValuesAreEqualToDefault() throws {
+		// GIVEN
+		let mockStore = MockTestStore()
+
+		let fileURL = try XCTUnwrap(Bundle(for: type(of: self)).url(forResource: "testData", withExtension: "json"))
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Scenes/Datadonation/__tests__/DataDonationModelTests.swift:14]: try 👏

**Review by marcussc — APPROVED** [review]

**Pull request merged**